File: VirtualGardens/VirtualGardens.Services/Database/PitanjaOdgovori.cs
using System;
using System.Collections.Generic;
using VirtualGardens.Services.BaseInterfaces;

namespace VirtualGardens.Services.Database;

public partial class PitanjaOdgovori : ISoftDeletable
{
    public int PitanjeId { get; set; }

    public string Tekst { get; set; } = null!;

    public DateTime Datum { get; set; } = DateTime.Now;

    public int KorisnikId { get; set; }

    public int NarudzbaId { get; set; }

    public virtual Korisnici Korisnik { get; set; } = null!;

    public virtual Narudzbe Narudzba { get; set; } = null!;
    public bool IsDeleted { get; set; } = false;
    public DateTime? VrijemeBrisanja { get; set; }
}

File: UI/virtualgardens_mobile/lib/models/pitanja_odgovori.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:virtualgardens_mobile/models/korisnici.dart';
import 'package:virtualgardens_mobile/models/narudzbe.dart';

part 'pitanja_odgovori.g.dart';

@JsonSerializable()
class PitanjeOdgovor {
  int pitanjeId;
  String tekst;
  DateTime datum;
  int korisnikId;
  int narudzbaId;
  Korisnik? korisnik;
  Narudzba? narudzba;

  PitanjeOdgovor({
    required this.pitanjeId,
    required this.tekst,
    required this.datum,
    required this.korisnikId,
    required this.narudzbaId,
    this.korisnik,
    this.narudzba,
  });

  factory PitanjeOdgovor.fromJson(Map<String, dynamic> json) =>
      _$PitanjeOdgovorFromJson(json);

  Map<String, dynamic> toJson() => _$PitanjeOdgovorToJson(this);
}
